src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AssemblyCache.netfx.cs

```diff
@@ -0,0 +1,44 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if NETFRAMEWORK
+
+using System;
+using System.Diagnostics;
+using Microsoft.Data.SqlClient.Server;
+
+namespace Microsoft.Data.SqlClient
+{
+    internal static class AssemblyCache
+    {
+        internal static int GetLength(object inst)
+        {
+            // Caller should have allocated enough, based on MaxByteSize
+            return SerializationHelperSql9.SizeInBytes(inst);
+        }
```

Comment:
This method is only called by SqlParameter.GetActualSize, [here](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlParameter.cs#L1617-L1622). We can remove the `#if` in SqlParameter, and delete the AssemblyCache.GetLength method completely.

Reply:
Fantastic catch! Yes that's a great change!!